picocrt/machine/arm/crt0.c

```diff
@@ -181,15 +194,16 @@ static const char *const reasons[] = {
 static void __attribute__((used))
 arm_fault(struct fault *f, int reason)
 {
-    printf("ARM fault: %s\n", reasons[reason]);
-    printf("\tR0:   0x%08x\n", f->r0);
-    printf("\tR1:   0x%08x\n", f->r1);
-    printf("\tR2:   0x%08x\n", f->r2);
-    printf("\tR3:   0x%08x\n", f->r3);
-    printf("\tR12:  0x%08x\n", f->r12);
-    printf("\tLR:   0x%08x\n", f->lr);
-    printf("\tPC:   0x%08x\n", f->pc);
-    printf("\tXPSR: 0x%08x\n", f->xpsr);
+    fputs("ARM fault: ", stdout);
+    puts(reasons[reason]);
```

Comment:
Let's use fputs for the reason too -- puts and fputs are completely separate implementations, so pulling in both doubles the amount of code used. That would require either adding a newline to each of the reasons or calling putchar separately; I'd try both and pick the one that results in a smaller result :-)  I thought I'd mentioned this in my earlier review; I must have been dreaming though. Sorry for the churn.

Reply:
The "put newlines in the string constants" approach won, but only by 4 bytes!